File: AutoSync/3095-shortest-subarray-with-or-at-least-k-i/solution.cpp
class Solution {
public:
    int minimumSubarrayLength(vector<int>& nums, int k) {
        return Method(nums,k);
    }
    int Method(vector<int>& nums,int k){
	int ans = INT_MAX;
	for(int i = 0;i < nums.size();i++){
		int n = 0;
		for(int j = i;j < nums.size();j++){
			n|=nums[j];
			if(n >= k){
				ans = min(ans,j-i+1);
				break;
			}
		}
	}
	return ans == INT_MAX?-1:ans;
}
};

File: AutoSync/3095-shortest-subarray-with-or-at-least-k-i/solution.java
class Solution {
    public int minimumSubarrayLength(int[] nums, int k) {
        return Method(nums,k);
    }
    public static int Method(int [] nums,int k){
        int count = Integer.MAX_VALUE;
        for (int i = 0; i < nums.length; i++) {
            int num = 0;
            for (int j = i; j < nums.length; j++) {
                num |= nums[j];
                if(num >= k){
                    count = Math.min(count,j-i+1);
                    break;
                }
            }
        }
         return count == Integer.MAX_VALUE?-1:count;
    }
}
